### PR TITLE
Implement a simple INI parser for config files

### PR DESCRIPTION
**Pull request by vvidic**
This removes the last runtime dependency on glib.

### CODE REVIEW

**Review by burgerdev — COMMENTED** [review]
Thanks for contributing this! 

Could you please add a paragraph to the readme documenting the config file syntax ("INI with # comments and without quoting and escaping"), and a hint to that paragraph to the example config?

File: login/config.c [hunk 1]
```diff
@@ -13,56 +13,191 @@
...
 #include "ui.h"
 
-int glome_login_parse_config_file(glome_login_config_t* config) {
-  g_autofree gchar* service_key = NULL;
-  guint64 service_key_version = 0;
-  g_autofree gchar* url_prefix = NULL;
-  g_autoptr(GKeyFile) cfg = g_key_file_new();
-  g_autoptr(GError) error = NULL;
+static bool is_empty(char *line) {
+  char *p;
```
[the remark this answers is not in the document]
Comment (vvidic) [reply login/config.c:24]: Done.

File: login/config.c [hunk 2]
```diff
@@ -13,56 +13,191 @@
...
 
 #include "ui.h"
 
-int glome_login_parse_config_file(glome_login_config_t* config) {
-  g_autofree gchar* service_key = NULL;
-  guint64 service_key_version = 0;
-  g_autofree gchar* url_prefix = NULL;
-  g_autoptr(GKeyFile) cfg = g_key_file_new();
-  g_autoptr(GError) error = NULL;
+static bool is_empty(char *line) {
```
[the remark this answers is not in the document]
Comment (vvidic) [reply login/config.c:23]: Done.

File: login/config.c [hunk 3]
```diff
@@ -13,56 +13,191 @@
...
+  }
+  if (*p != ']') {
+    return NULL;
+  }
+  char *end = p;
+  for (p++; isspace(*p); p++) {
+  }
+  if (*p != '\0') {
+    return NULL;
+  }
```
[the remark this answers is not in the document]
Comment (vvidic) [reply login/config.c:55]: Done.

File: login/config.c [hunk 4]
```diff
@@ -13,56 +13,191 @@
...
+    }
+  } else if (strcmp(key, "key-version") == 0) {
+    char *end;
+    errno = 0;
+    long n = strtol(val, &end, 10);
+    if (errno != 0 || *end != '\0') {
+      errorf("ERROR: Failed to parse service key version\n");
+      return false;
+    }
+    if (n > 255) {
```
[the remark this answers is not in the document]
Comment (vvidic) [reply login/config.c:128]: Done.

**Comment by vvidic** [comment]
Thank you for the review! Readme and example config are now updated too.

**Review by burgerdev — APPROVED** [review]

**Pull request merged**